frontend/svelte/jest-setup.ts

```diff
@@ -1,7 +1,32 @@
 import "@testing-library/jest-dom";
-
 // jsdom does not implement TextEncoder
 // Polyfill the encoders with node
-import { TextEncoder, TextDecoder } from "util";
+import { TextDecoder, TextEncoder } from "util";
+
 global.TextEncoder = TextEncoder;
 global.TextDecoder = TextDecoder;
+
+// @ts-ignore
```

Comment:
```suggestion
// Mocks the Intersection Observer API for the tests. This just matches the interface
// but doesn't do anything.
// https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
// @ts-ignore
```

note: do we actually test the behavior?

Reply:
yes. the component won't compile without it and it tests the the `nnsIntersect` event is dispatched by the component if the intersection observer calls the callback with an intersecting elements.

being said, as I described in the PR details, it will need e2e tests.